### PR TITLE
Add operators support

### LINKED ISSUE #114: Operators are not being tokenized
They might be not tagged because operators can be overloaded. Not sure about that.
See [language-javascript#L966](https://github.com/atom/language-javascript/blob/9cf9a3970369b247878e98897ce1549b94769f38/grammars/javascript.cson#L966), [javascript-spec.coffee#L252](https://github.com/atom/language-javascript/blob/82fffc468f995feec44f1b1581e6ba8ac95fb642/spec/javascript-spec.coffee#L252) as an example

List of operators: https://en.wikipedia.org/wiki/Operators_in_C_and_C++#Table

### LINKED ISSUE #123: Operators like + , -, *, /, %, ^, |, &, ||, && Highlighting?
Is there any option to highlight operators like + , -, *, /, %, ^, |, &, ||, && in C++?

### PR DESCRIPTION
**Pull request by MaximSokolov**
Fixes #114, fixes #123
```c
// sizeof
sizeof unary_expression
sizeof(int)

// arithmetic
+a
-a

a + b
a - b
a * b
a / b
a % b

i++
++i
i--
--i

// comparison
a < b
a > b
a >= b
a <= b
a == b
a != b

// bitwise
~a
a & b
a | b
a ^ b
a << b
a >> b

// logical
!a
a && b
a || b

// assignment
a = b
a += b
a -= b
a *= b
a /= b
a %= b
a &= b
a ^= b
a |= b
a <<= b
a >>= b

// pointer
*a
&a

// ternary
a ? b : c
```

### CODE REVIEW

File: grammars/c.cson [hunk 1]
```diff
@@ -701,9 +701,68 @@
...
+        'match': '%=|\\+=|-=|\\*=|(?<!\\()/='
+        'name': 'keyword.operator.assignment.compound.c'
+      }
+      {
+        'match': '&=|\\^=|<<=|>>=|\\|='
+        'name': 'keyword.operator.assignment.compound.bitwise.c'
+      }
+      {
+        'match': '<<|>>'
+        'name': 'keyword.operator.c'
```
Comment (50Wliu) [inline grammars/c.cson]: maybe `keyword.operator.bitwise.c`?

File: grammars/c.cson [hunk 2]
```diff
@@ -701,9 +701,68 @@
...
+        'match': '!=|<=|>=|==|<|>'
+        'name': 'keyword.operator.comparison.c'
+      }
+      {
+        'match': '&&|!|\\|\\|'
+        'name': 'keyword.operator.logical.c'
+      }
+      {
+        'match': '&|\\||\\^|~'
+        'name': 'keyword.operator.c'
```
Comment (50Wliu) [inline grammars/c.cson]: .logical.bitwise?

File: spec/c-spec.coffee [hunk 3]
```diff
@@ -527,6 +536,112 @@ describe "Language-C", ->
         expect(lines[2][1]).toEqual value: '->', scopes: ['source.c', 'meta.function.c', 'meta.block.c', 'punctuation.separator.pointer-access.c']
         expect(lines[2][2]).toEqual value: 'b', scopes: ['source.c', 'meta.function.c', 'meta.block.c', 'variable.other.member.c']
 
+    describe "operators", ->
+      it "tokenizes the sizeof operator", ->
```
Comment (50Wliu) [inline spec/c-spec.coffee]: Can we throw in a `sizeof$` or a `sizeof_` check somewhere?

**Comment by 50Wliu** [comment]
Very nice work as usual @MaximSokolov.  Left some pretty minor comments.

File: grammars/c.cson (hunk 2 above)
Comment (MaximSokolov) [inline grammars/c.cson]: It's not possible to distinguish an indirection operator and a multiplication operator
```c
a * b
*a
```
Same for bitwise NOT and address-of operators
```c
&a
a & b
```
I didn't add `bitwise` subscope for `|`, `~`, `^` for consistency with `&`. I made an exception for shift operators(see 2a5fafe)

**Comment by MaximSokolov** [comment]
The only issue with overloading operators:
```cpp
cout << "test"; // insertion operator '<<' would have a `bitwise.shift` subscope
```
Fixed in 	80db385:
```cpp
~Foo() {}
```
I didn't fix this:
```cpp
vector<vector<int> > a;
```
```cpp
template <typename T>
```

File: grammars/c.cson (hunk 2 above)
Comment (50Wliu) [inline grammars/c.cson]: Good point.  This isn't something that the grammar can handle right now.

**Comment by 50Wliu** [comment]
So while there are some incorrect scopes introduced with this PR, I think that the benefits of at least being able to syntax highlight operators outweighs the downsides of said incorrect scopes.

@MaximSokolov would you be willing to create new issues for your last two examples?

**Pull request merged**